src/library/dlgtrackinfo.cpp

```diff
@@ -384,54 +479,29 @@ void DlgTrackInfo::saveTrack() {
 
     // First, disconnect the track changed signal. Otherwise we signal ourselves
     // and repopulate all these fields.
-    disconnect(m_pLoadedTrack.get(),
-            &Track::changed,
-            this,
-            &DlgTrackInfo::slotTrackChanged);
+    const QSignalBlocker signalBlocker(this);
 
-    m_pLoadedTrack->setTitle(txtTrackName->text());
-    m_pLoadedTrack->setArtist(txtArtist->text());
-    m_pLoadedTrack->setAlbum(txtAlbum->text());
-    m_pLoadedTrack->setAlbumArtist(txtAlbumArtist->text());
-    m_pLoadedTrack->setGenre(txtGenre->text());
-    m_pLoadedTrack->setComposer(txtComposer->text());
-    m_pLoadedTrack->setGrouping(txtGrouping->text());
-    m_pLoadedTrack->setYear(txtYear->text());
-    m_pLoadedTrack->setTrackNumber(txtTrackNumber->text());
-    m_pLoadedTrack->setComment(txtComment->toPlainText());
-
-    m_pLoadedTrack->trySetBeats(m_pBeatsClone);
-    reloadTrackBeats(*m_pLoadedTrack);
-
-    // If the user is editing the key and hits enter to close DlgTrackInfo, the
-    // editingFinished signal will not fire in time. Run the key text changed
-    // handler now to see if the key was edited. If the key was unchanged or
-    // invalid then the change will be rejected.
-    slotKeyTextChanged();
-
-    m_pLoadedTrack->setKeys(m_keysClone);
-    m_pLoadedTrack->setCoverInfo(m_loadedCoverInfo);
-
-    // Reconnect changed signals now.
-    connect(m_pLoadedTrack.get(),
-            &Track::changed,
-            this,
-            &DlgTrackInfo::slotTrackChanged);
-}
+    // Special case handling for the comment field that is not
+    // updated by the editingFinished signal.
+    m_trackRecord.refMetadata().refTrackInfo().setComment(txtComment->toPlainText());
```

Comment:
Why is it not updated by the editingFinished signal?

Reply:
This has not changed. Because QPlainTextEdit only has a textChanged() but no editingFinished() signal?

Reply:
OK, I was just wondering.